sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusMessageSerializer.java

```diff
@@ -225,6 +237,53 @@ public int getSize(Message amqpMessage) {
         return amqpMessage;
     }
 
+    /**
+     * For send application properties with OffsetDateTime, Duration and URI on the wire in amqp,
+     * we need to convert these object to described type and then amqp can write these data into buffer.
+     * @param propertiesValue application properties set by user which may contain specific type mentioned above.
+     * @return Map only contains primitive type and described type.
+     */
+    private static Map<String, Object> convertToDescribedType(Map<String, Object> propertiesValue) {
+        for (Map.Entry<String, Object> entry : propertiesValue.entrySet()) {
+            Object value = entry.getValue();
+            if (value instanceof URI) {
+                propertiesValue.put(entry.getKey(), new UriDescribedType(value));
+            } else if (value instanceof OffsetDateTime) {
+                propertiesValue.put(entry.getKey(), new OffsetDateTimeDescribedType(value));
+            } else if (value instanceof Duration) {
+                propertiesValue.put(entry.getKey(), new DurationDescribedType(value));
+            }
+        }
+        return propertiesValue;
+    }
+
+    /**
+     * Reverse convert ServiceBusMessageSerializer#convertToDescribedType. If convert failed, put back the value.
+     * @param propertiesValue application properties from amqp message may contain described type.
+     * @return Map without described type.
+     */
+    private static Map<String, Object> convertToOriginType(Map<String, Object> propertiesValue) {
+        for (Map.Entry<String, Object> entry : propertiesValue.entrySet()) {
+            Object value = entry.getValue();
+            if (value instanceof DescribedType) {
+                try {
+                    DescribedType describedType = (DescribedType) value;
+                    if (ServiceBusConstants.URI_SYMBOL.equals(describedType.getDescriptor())) {
+                        propertiesValue.put(entry.getKey(), describedToURI(describedType.getDescribed()));
```

Comment:
Should it be `entry.setValue`?

Modifying the Map during its iteration is problematic.

Reply:
Sure, I'll fix this.